pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -167,51 +313,425 @@ func certsTask(ctx *zedagentContext, triggerCerts <-chan struct{}) {
+			log.Errorf("sendEveNodeCertsProtobuf failed: %s", err)
 		}
-		// Try sending later
-		// The buf might have been consumed
-		buf := bytes.NewBuffer(data)
-		if buf == nil {
-			log.Fatal("malloc error")
-		}
-		zedcloud.SetDeferred("attest:"+deviceUUID, buf, size, attestURL,
+		zedcloud.SetDeferred("attest:"+reqId, buf, size, attestURL,
 			zedcloudCtx, true)
 	}
 }
+
+// look up controller certificate config
+func lookupControllerCertConfig(ctx *cipherContext,
+	key string) *types.ControllerCertConfig {
+	log.Infof("lookupControllerCertConfig(%s)", key)
+	pub := ctx.pubControllerCertConfig
+	item, err := pub.Get(key)
+	if err != nil {
+		log.Errorf("lookupControllerCertConfig(%s) not found", key)
+		return nil
+	}
+	config := item.(types.ControllerCertConfig)
+	log.Infof("lookupControllerCertConfig(%s) Done", key)
+	return &config
+}
+
+// look up controller certificate status
+func lookupControllerCertStatus(ctx *cipherContext,
+	key string) *types.ControllerCertStatus {
+	log.Infof("lookupControllerCertStatus(%s)", key)
+	pub := ctx.pubControllerCertStatus
+	item, err := pub.Get(key)
+	if err != nil {
+		log.Errorf("lookupControllerCertStatus(%s) not found", key)
+		return nil
+	}
+	status := item.(types.ControllerCertStatus)
+	log.Infof("lookupControllerCertStatus(%s) Done", key)
+	return &status
+}
+
+// look up eve node certificate status
+func lookupEveNodeCertStatus(ctx *cipherContext,
+	key string) *types.EveNodeCertStatus {
+	log.Infof("lookupEveNodeCertStatus(%s)", key)
+	sub := ctx.subEveNodeCertStatus
+	item, err := sub.Get(key)
+	if err != nil {
+		log.Errorf("lookupEveNodeCertStatus(%s) not found", key)
+		return nil
+	}
+	status := item.(types.EveNodeCertStatus)
+	log.Infof("lookupEveNodeCertStatus(%s) Done", key)
+	return &status
+}
+
+// look up cipher context status
+func lookupCipherContextStatus(ctx *cipherContext,
+	key string) *types.CipherContextStatus {
+	log.Infof("lookupCipherContextStatus(%s)", key)
+	sub := ctx.subCipherContextStatus
+	item, err := sub.Get(key)
+	if err != nil {
+		log.Errorf("lookupCipherContextStatus(%s) not found", key)
+		return nil
+	}
+	status := item.(types.CipherContextStatus)
+	log.Infof("lookupCipherContextStatus(%s) Done", key)
+	return &status
+}
+
+// Controller certificate, check whether there is a Sha mismatch
+// to trigger the post request
+func handleControllerCertsSha(ctx *zedagentContext,
+	config *zconfig.EdgeDevConfig) {
+	sumHash := hex.EncodeToString(controllerCertsHash)
+	certHash := config.GetControllercertConfighash()
+	if sumHash != certHash {
+		triggerControllerCertsEvent(ctx)
+	}
+}
+
+// Eve node certificate, check whether there is a Sha mismatch
+// to trigger the post request
+func handleEveNodeCertsSha(ctx *zedagentContext,
+	config *zconfig.EdgeDevConfig) {
+	sumHash := hex.EncodeToString(eveNodeCertsHash)
+	certHash := config.GetNodecertConfighash()
+	if sumHash != certHash {
+		triggerEveNodeCertsEvent(ctx)
+	}
+}
```

Comment:
Why do we need this?
The publish from tpmmgr will only publish if there is a change, hence when handleModify is called in zedagent we know that there was a change.